services/agent/src/simcore_service_agent/modules/task_monitor.py

```diff
@@ -0,0 +1,140 @@
+import asyncio
+import logging
+import traceback
+from collections import deque
+from contextlib import suppress
+from dataclasses import dataclass
+from typing import Any, Awaitable, Callable, Coroutine, Final, Optional
+
+from fastapi import FastAPI
+from pydantic import PositiveFloat, PositiveInt
+
+from ..core.settings import ApplicationSettings
+from .volumes_cleanup import backup_and_remove_volumes
+from servicelib.logging_utils import log_context
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_TASK_WAIT_ON_ERROR: Final[PositiveInt] = 10
+
+
+@dataclass
+class _TaskData:
+    callable: Callable
+    args: Any
+    repeat_interval_s: Optional[PositiveFloat]
+
+
+async def _task_runner(task_data: _TaskData) -> None:
+    with log_context(logger, logging.INFO, msg=f"'{task_data.callable.__name__}'"):
+        while True:
+            coroutine: Coroutine = task_data.callable(*task_data.args)
```

Comment:
this also reads very redundant ... you already annotated with ``Coroutine``, why to repeat it? 
It reminds me something like
```python
_int: int  = 3
_float: float = 3.0
_string: string = "one"
```
we loose the advantage that we were given withthe annotation. When there is annotation, we should add other info in the name of the variable

What about
```python
while True:
     try:
          await task_data.start()  # or await task_data.coro(*task_data.args)
     except Exception:
          logger.error( " %s", task_data.coro.__name__ )
```

Reply:
Good point, simplified the entire thing quite a bit.